paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziViewOwners.kt

```diff
@@ -0,0 +1,50 @@
+package app.cash.paparazzi.internal
+
+import android.view.View
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.savedstate.SavedStateRegistry
+import androidx.savedstate.SavedStateRegistryController
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.setViewTreeSavedStateRegistryOwner
+
+internal object PaparazziViewOwners {
+  private val hasLifecycleOwnerRuntime = isPresentInClasspath("androidx.lifecycle.LifecycleOwner")
+  private val hasSavedStateRegistryOwnerRuntime = isPresentInClasspath("androidx.savedstate.SavedStateRegistryOwner")
+
+  fun register(view: View) {
+    if (hasLifecycleOwnerRuntime) {
+      val lifecycleOwner = PaparazziLifecycleOwner()
+      ViewTreeLifecycleOwner.set(view, lifecycleOwner)
+
+      if (hasSavedStateRegistryOwnerRuntime) {
+        view.setViewTreeSavedStateRegistryOwner(PaparazziSavedStateRegistryOwner(lifecycleOwner))
+      }
+
+      lifecycleOwner.registry.currentState = Lifecycle.State.CREATED
```

Comment:
nit: should this move to before line 20 (maintaining the previous setup/execution order) or does it not matter?

Reply:
The lifecycle state can't be changed from `INITIALIZED` until `SavedStateRegistryOwner` has finished restoring all state. I've added a comment. Let me know if you think that our failing tests were sufficient?